src/sedtrails/transport_converter/format_converter.py

```diff
@@ -1,80 +1,544 @@
+                'magnitude': self.depth_avg_flow_velocity['magnitude'][time_idx]
+            },
+            'bed_load_sediment': {
+                'x': self.bed_load_sediment['x'][time_idx],
+                'y': self.bed_load_sediment['y'][time_idx],
+                'magnitude': self.bed_load_sediment['magnitude'][time_idx]
+            },
+            'suspended_sediment': {
+                'x': self.suspended_sediment['x'][time_idx],
+                'y': self.suspended_sediment['y'][time_idx],
+                'magnitude': self.suspended_sediment['magnitude'][time_idx]
+            },
+            'nonlinear_wave_velocity': {
+                'x': self.nonlinear_wave_velocity['x'][time_idx],
+                'y': self.nonlinear_wave_velocity['y'][time_idx],
+                'magnitude': self.nonlinear_wave_velocity['magnitude'][time_idx]
+            }
+        }
+        
+        return time_data
+    
+    def __len__(self) -> int:
+        """
+        Get the number of time steps.
+        
+        Returns:
+        --------
+        int
+            Number of time steps
+        """
+        return len(self.times)
+    
+    def get_time_index(self, target_time: float) -> int:
+        """
+        Find the index of the closest time to the target time.
+        
+        Parameters:
+        -----------
+        target_time : float
+            Target time in seconds since reference_date
+            
+        Returns:
+        --------
+        int
+            Index of the closest time
+        """
+        # Find the closest time
+        time_diffs = np.abs(self.times - target_time)
+        return np.argmin(time_diffs)
+    
+    def get_interpolation_indices(self, target_time: float) -> Tuple[int, int, float]:
+        """
+        Get indices for interpolation between two time steps.
+        
+        Parameters:
+        -----------
+        target_time : float
+            Target time in seconds since reference_date
+            
+        Returns:
+        --------
+        Tuple[int, int, float]
+            (lower_index, upper_index, weight) where weight is the interpolation factor [0-1]
+        """
+        if target_time <= self.times[0]:
+            return 0, 0, 0.0
+        
+        if target_time >= self.times[-1]:
+            return len(self.times) - 1, len(self.times) - 1, 1.0
+        
+        # Find the index of the last time that is less than or equal to the target time
+        lower_idx = np.searchsorted(self.times, target_time, side='right') - 1
+        upper_idx = lower_idx + 1
+        
+        # Calculate the interpolation weight
+        time_range = self.times[upper_idx] - self.times[lower_idx]
+        
+        # Avoid division by zero
+        if time_range == 0:
+            weight = 0.0
+        else:
+            weight = (target_time - self.times[lower_idx]) / time_range
+            
+        return lower_idx, upper_idx, weight
```

Comment:
Isn't this code repeated in  `data_retriever.py`? If we need it in both, that OK. We just have to make sure that the code is not executed twice for the same operation.